x-pack/plugins/fleet/public/applications/fleet/sections/agent_policy/components/agent_policy_create_inline.tsx

```diff
@@ -0,0 +1,187 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { useCallback, useState } from 'react';
+import {
+  EuiAccordion,
+  EuiButton,
+  EuiFieldText,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiForm,
+  EuiFormRow,
+  EuiLink,
+  EuiSpacer,
+  EuiText,
+} from '@elastic/eui';
+import { FormattedMessage } from '@kbn/i18n-react';
+import styled from 'styled-components';
+import { i18n } from '@kbn/i18n';
+
+import { dataTypes } from '../../../../../../common';
+import { agentPolicyFormValidation } from '../components';
+
+import type { AgentPolicy, NewAgentPolicy } from '../../../types';
+
+import { sendCreateAgentPolicy } from '../../../hooks';
+
+import { AgentPolicyAdvancedOptionsContent } from './agent_policy_advanced_fields';
+import { AgentPolicyFormSystemMonitoringCheckbox } from './agent_policy_system_monitoring_field';
+
+const StyledEuiAccordion = styled(EuiAccordion)`
+  .ingest-active-button {
+    color: ${(props) => props.theme.eui.euiColorPrimary};
+  }
+`;
+
+interface Props {
+  updateAgentPolicy: (u: AgentPolicy | null) => void;
+  isFleetServerPolicy?: boolean;
+  agentPolicyName: string;
+}
+
+export const AgentPolicyCreateInlineForm: React.FunctionComponent<Props> = ({
+  updateAgentPolicy,
+  isFleetServerPolicy,
+  agentPolicyName,
+}) => {
+  const [touchedFields, setTouchedFields] = useState<{ [key: string]: boolean }>({});
+
+  const [withSysMonitoring, setWithSysMonitoring] = useState<boolean>(true);
+
+  const [isLoading, setIsLoading] = useState(false);
+
+  const [newAgentPolicy, setNewAgentPolicy] = useState<NewAgentPolicy>({
+    name: agentPolicyName,
+    description: '',
+    namespace: 'default',
+    monitoring_enabled: Object.values(dataTypes),
+    has_fleet_server: isFleetServerPolicy,
+  });
+
+  const updateNewAgentPolicy = useCallback(
+    (updatedFields: Partial<NewAgentPolicy>) => {
+      setNewAgentPolicy({
+        ...newAgentPolicy,
+        ...updatedFields,
+      });
+    },
+    [setNewAgentPolicy, newAgentPolicy]
+  );
+
+  const validation = agentPolicyFormValidation(newAgentPolicy);
+
+  const createAgentPolicy = useCallback(async () => {
+    try {
+      setIsLoading(true);
+      const resp = await sendCreateAgentPolicy(newAgentPolicy, { withSysMonitoring });
+      if (resp.error) throw resp.error;
+      if (resp.data) {
+        updateAgentPolicy(resp.data.item);
+      }
+    } catch (e) {
+      updateAgentPolicy(null);
```

Comment:
Should we display a toast to show the error if an error happen?

Reply:
it is done with a callout (see UX design)
![image](https://user-images.githubusercontent.com/90178898/151332386-0504175d-dd5a-47f2-ba1a-a2ff137639f2.png)
https://github.com/elastic/kibana/pull/121628/files/62c5f2346abc20df2230b568bab0ba6cf221a60d#diff-8f29f5db178ab8f5016450c151aba59bb72a530e1332d093c54212e33ea5da2c